core/schema/secret.go

```diff
@@ -25,19 +26,26 @@ func (s *secretSchema) Install() {
 			ArgSensitive("plaintext"),
 
 		dagql.Func("secret", s.secret).
-			Doc(`Reference a secret by name.`),
+			Impure("`secret` mutates state in the internal secret store.").
+			Doc(`Creates a new secret.`).
+			ArgDoc("uri", `The URI of the secret store`),
+
+		dagql.Func("loadSecretFromName", s.loadSecretFromName).
```

Comment:
Just making a note that https://github.com/dagger/dagger/pull/9204 should allow us to rm `loadSecretFromName`.

I thought I squashed them, but appears to be some memory leak that still needs a fix there, so feel free to merge this PR whenever ready and I can take care of rm'ing `loadSecretFromName` when rebasing. Any order of operations is fine by me.

---

The basic plan would be to leave `secret` impure but in the implementation return a pure instance that has a custom cache key where the `Accessor` is mixed in.